content/blog/2025-07-27-utxo-redemption.md

```diff
@@ -0,0 +1,59 @@
++++
+title = "UTXO Redemption"
+
+[extra]
+author = "Alan Szepieniec"
+ogdescription = "Details about the UTXO redemption procedure"
+ogimage = "utxo-redemption-diagram.jpg"
+ogtype = "article"
++++
+
+| ![Schematic diagram of a UTXO redeeming itself from another chain.](utxo-redemption-diagram.jpg) |
+| :-----------------: |
+| **Fig.1.**: Grok adamantly refusing to imagine an abstract cross-chain UTXO redemption protocol. |
+
+# UTXO Redemption
+
+On July 4 2025, Neptune-core developers discovered a catastrophic [undetectable inflation bug](../inflation-bug-discovered). Following this discovery, the decision was made to reboot the blockchain with a new genesis block. It is not a pure reboot though, because there is a UTXO redemption claims process whereby owners of UTXOs on the old chain can reclaim their balance on the new chain.
+
+## The Claims Fund
+
+Specifically, the block subsidy schedule on the reboot chain is fast-forwarded by 21310 blocks, meaning that the first halving will occur 21310 blocks sooner. The skipped block subsidies are be allocated to a claims fund from which disbursements can be made to parties who can prove ownership of UTXOs at block 21310 on the old chain. In this way, the 42'000'000 coins limit remains.
+
+The claims fund will initially be custodied by the founders. After a while, when the infrastructure exists to support it, they may be moved to a smart contract that can validate the redemption claims and automatically make the disbursements when the claims are found valid. Alternatively, they will be burned. We have not decided yet.
+
+If the vulnerability was exploited before block 21310 (which marks the time the bug was discovered), then the exploiter will be able to produce a UTXO redemption claim that is indistinguishable from authentic. If they choose to exercise this claim, it is possible that the fund runs out before other, legitimate, users have had a chance to exercise their claims. In this case, the fund abides by the "first come, first serve" principle because serving all claimants would undermine the 42'000'000 limit and the only alternative is to pick winners and losers based on different information from which claims are legitimate and which are not.
```

Comment:
Consider adding: "It's worth noting that our best guess is that the bug has not been exploited. But we cannot give any guarantees that that's the case."